pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -435,9 +412,58 @@ public void addSegment(ImmutableSegment segment) {
     }
   }
 
+  protected boolean isTTLEnabled() {
+    return _metadataTTL > 0 || _deletedKeysTTL > 0;
+  }
+
+  protected void updateTTLWatermark(ImmutableSegment segment) {
```

Comment:
Actually we are just updating `_largestSeenComparisonValue` here and not persisting back to watermark. The naming might be confusing.

Reply:
moved the this method close to other watermark related methods persist/load/deleteWatermark()

I take `_largestSeenComparisonValue` same as watermark.